app/src/main/java/com/mapbox/navigation/examples/junctions/ShowJunctionsActivity.kt

```diff
@@ -0,0 +1,238 @@
+ *
+ * Note: A special access token is required to get access to junctions in directions response.
+ */
+class ShowJunctionsActivity : AppCompatActivity() {
+
+    private lateinit var mapboxMap: MapboxMap
+    private lateinit var mapboxNavigation: MapboxNavigation
+    private lateinit var binding: MapboxActivityShowJunctionsBinding
+    private lateinit var locationComponent: LocationComponentPlugin
+
+    private val route by lazy {
+        DirectionsRoute.fromJson(String(assets.open("hardcoded_junctions_route.json").use { it.readBytes() }))
+            .toNavigationRoute()
+    }
+    private val mapboxReplayer = MapboxReplayer()
+    private val navigationLocationProvider = NavigationLocationProvider()
+
+    /**
+     * The [MapboxJunctionApi] consumes banner instructions data and produces junctions related
+     * data that is consumed by the [MapboxJunctionView] in the view layout.
+     *
+     * Uses a specific access token required for the route request to send junctions in the response.
+     */
+    private val junctionApi: MapboxJunctionApi by lazy {
+        MapboxJunctionApi(getString(R.string.mapbox_access_token))
+    }
+
+    /**
+     * The result of invoking [MapboxJunctionApi.generateJunction] is returned as a callback
+     * containing either a success in the form of [JunctionValue] or failure in the form of
+     * [JunctionError].
+     */
+    private val junctionCallback =
+        MapboxNavigationConsumer<Expected<JunctionError, JunctionValue>> { value ->
+            // The data obtained must be rendered by [MapboxJunctionView]
+            binding.junctionView.render(value)
+        }
+
+    private val routeLineResources: RouteLineResources by lazy {
+        RouteLineResources.Builder().build()
+    }
+
+    private val options: MapboxRouteLineOptions by lazy {
+        MapboxRouteLineOptions.Builder(this)
+            .withRouteLineResources(routeLineResources)
+            .withRouteLineBelowLayerId("road-label")
+            .build()
+    }
+
+    private val routeLineView by lazy {
+        MapboxRouteLineView(options)
+    }
+
+    private val routeLineApi: MapboxRouteLineApi by lazy {
+        MapboxRouteLineApi(options)
+    }
+
+    private val replayProgressObserver = ReplayProgressObserver(mapboxReplayer)
+
+    private val locationObserver = object : LocationObserver {
+        override fun onNewRawLocation(rawLocation: Location) {}
+        override fun onNewLocationMatcherResult(locationMatcherResult: LocationMatcherResult) {
+            navigationLocationProvider.changePosition(
+                locationMatcherResult.enhancedLocation,
+                locationMatcherResult.keyPoints,
+            )
+            updateCamera(locationMatcherResult.enhancedLocation)
+        }
+    }
+
+    private val routesObserver = RoutesObserver { result ->
+        if (result.navigationRoutes.isNotEmpty()) {
+            CoroutineScope(Dispatchers.Main).launch {
```

Comment:
```suggestion
            lifecycleScope.launch {
```
Otherwise this coroutine won't be canceled if activity finishes before completion.